test/jasmine/assets/click.js

```diff
@@ -43,10 +45,16 @@ module.exports = function click(x, y, optsIn) {
 
     var downOpts = Lib.extendFlat({buttons: buttons}, opts);
     var upOpts = opts;
+    var slop = opts.slop || [];
+    var slopX = slop[0] || 0;
+    var slopY = slop[1] || 0;
+    var slopOpts = Lib.extendFlat({}, downOpts);
+    delete slopOpts.button;
 
     mouseEvent('mousemove', x, y, moveOpts);
     mouseEvent('mousedown', x, y, downOpts);
     if(callContext) mouseEvent('contextmenu', x, y, downOpts);
+    if(slopX || slopY) mouseEvent('mousemove', x + slopX, y + slopY, slopOpts);
```

Comment:
So, `slop` is only used for testing? This is fine, but perhaps we could add a comment about this in the jsDoc above so that we don't feel like :hocho: down the road.

Reply:
sure -> https://github.com/plotly/plotly.js/pull/2649/commits/c56f223fd0aff569ac9c2da3f2ae79957dab8908